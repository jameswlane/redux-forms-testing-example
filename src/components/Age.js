import React from 'react'
import { Field } from "redux-form";
import { RenderField } from "./RenderField";
import { minValue, number, required, tooYoung } from "../validation";

export function Age() {
  return <Field
    name="age"
    type="number"
    component={RenderField}
    label="Age"
    validate={[required, number, minValue]}
    warn={tooYoung}
  />;
}
