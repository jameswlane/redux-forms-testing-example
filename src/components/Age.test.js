import React from 'react'
import { cleanup } from 'react-testing-library'

import { Age } from './Age';
import { renderReduxForm } from '../utilities/render-redux-form';

describe('Component => Age', () => {
  afterEach(cleanup);
  it('should match snapshot', () => {
    const { container } = renderReduxForm(Age);

    expect(container.firstChild).toMatchSnapshot();
  })
});


