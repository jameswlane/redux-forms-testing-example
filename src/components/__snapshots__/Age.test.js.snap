// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component => Age should match snapshot 1`] = `
<form>
  <div>
    <label>
      Age
    </label>
    <div>
      <input
        name="age"
        placeholder="Age"
        type="number"
        value=""
      />
    </div>
  </div>
</form>
`;
